hook_macro/src/lib.rs

```diff
@@ -0,0 +1,72 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, AttributeArgs};
+use quote::ToTokens;
+
+
+#[proc_macro_attribute]
+/// This macro is used to insert a hook at the start of a function.
+pub fn cvt_hook_start(attr: TokenStream, input: TokenStream) -> TokenStream {
+    
+    // parse the attribute argument
+    let attr = parse_macro_input!(attr as AttributeArgs);
+    if attr.len() != 1 {
+        panic!("expected 1 argument");
+    }
+
+    let arg = &attr[0]; 
+
+    // parse the input tokens and make sure it is a function
+    let mut item: syn::Item = syn::parse(input).unwrap();
+    let fn_item = match &mut item {
+        syn::Item::Fn(fn_item) => fn_item,
+        _ => panic!("expected fn")
+    };
+
+
+    // insert tokens_start into fn item statements at position 0
+    let tokens_start = quote! {
+        if cfg!(feature = "certora") {
```

Comment:
this is not needed. We can condition the hook attribute itself by a feature. There is no need to generate code that itself depends on features since the value of the `cfg!` expression is known at compile time.